doc/designs/dtlsv1_3/dtlsv1_3-main.md

```diff
@@ -70,6 +70,26 @@ section 5.9.
 The DTLSv1.3 implementation modifies the epoch according to RFC9147 section 6.1
 for DTLSv1.3 connections.
 
+#### DTLS ACK records (RFC9147 Section 7)
+
+ACK's are sent for KeyUpdates, NewSessionTicket and Finish (client).
+
+Notes on RFC9147 Section 7.1:
+
+* The implementation does not offer any logic to determine that there is disruption
+  when receiving messages which means it will not send ACKs for the example given
+  in RFC9147 Figure 12.
+* ACKs are always sent immediately after receiving a full message to be ACK'ed.
+* If the implementation does not receive an ACK for all fragments of a message,
+  then the full message wil be retransmitted.
+* Empty ACKs are never sent.
+* The implementation does not explicitly prohibit receiving unencrypted ACKs. The
+  implementation will only ACK records of epoch > 0 so all ACKs sent by the
+  implementation will be encrypted.
+* The implementation only accepts ACKs after DTLSv1.3 has been negotiated. ACKs
+  that are received when DTLSv1.3 has not been negotiated is handled with a fatal
+  alert as any other unexpected message.
```

Comment:
This is a little subtle. Keep in mind that the server learns the final version before the client does, and we have re-ordering. See this thread:
https://mailarchive.ietf.org/arch/msg/tls/ZEj04LyL3hJXeK1nsiOBoB2vCsg/

The server could plausibly send ServerHello and a cleartext ACK but the ServerHello is lost. Now, I suspect we could fix the protocol so that this is not possible, because it's kind of a silly case. If the server takes too long to generate CertificateVerify, there honestly isn't a whole lot they can do, if they flush a whole flight at a time, which is what the protocol generally encourages.

But the document is also not very prescriptive about ACK-sending behavior. Given this mess, I think it's safer for the client to tolerate (but probably drop) ACKs before the version is known.

Reply:
I see your point. My suggestion would be to let it depend on whether DTLS 1.3 has been offered by the client.

A client offering DTLS 1.3 and receiving a plaintext acknowledge drops it.
A client that has not offered DTLS 1.3 and receives an ack sends a fatal alert.

Reply:
Yeah, that seems reasonable. That's what we're doing for BoringSSL. (Well, we opted to just always drop it if the version is not known, just for simplicity. TBH, there's probably an argument that we should be dropping all unknown record types, but DTLS is very ambiguous about what "invalid records" means when it says to drop them.)

Reply:
Done.